<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSII/iLogos/logo_novafct.png" width="200">

# Departamento de Engenharia Mecânica e Industrial
## Mecânica dos Sólidos II

## Tensões e deformações plásticas

### Problema 1

Considere a viga representada na figura sujeita à flexão apenas em torno do eixo horizontal. Sabendo que $\sigma_\mathrm{ced}$ = 250 MPa, determine o momento máximo elástico, $M_\mathrm{C}$, o momento plástico, $M_\mathrm{P}$, e o fator de forma, $K$, da sua secção transversal.

<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSII/Notebooks/Au03/P1/MSII_Au03_P1.png"
width="400">

## Resolução

### Dados do problema:

In [1]:
import numpy as np
import sympy as sy
from sympy.solvers import solve
import matplotlib.pyplot as plt
import os

cor = '2'
if cor == '1':
    plt.rcParams['axes.facecolor'] = (.15, .15, .15)
    plt.rcParams['figure.facecolor'] = (.15, .15, .15)
    plt.rcParams['font.family'] = 'monospace'
    plt.rcParams['font.size'] = 18
    # plt.rcParams['text.usetex'] = True
    params = {"ytick.color" : (.8, .8, .8),
              "xtick.color" : (.8, .8, .8),
              "grid.color" : (.2, .2, .2),
              "text.color" : (.7, .7, .7),
              "axes.labelcolor" : (.8, .8, .8),
              "axes.edgecolor" : (.15, .15, .15)}
else:
    plt.rcParams['axes.facecolor'] = (.7, .7, .7)
    plt.rcParams['figure.facecolor'] = (.7, .7, .7)
    plt.rcParams['font.family'] = 'monospace'
    plt.rcParams['font.size'] = 18
    # plt.rcParams['text.usetex'] = True
    params = {"ytick.color" : (.1, .1, .1),
              "xtick.color" : (.1, .1, .1),
              "grid.color" : (.2, .2, .2),
              "text.color" : (.1, .1, .1),
              "axes.labelcolor" : (.1, .1, .1),
              "axes.edgecolor" : (.15, .15, .15)}
plt.rcParams.update(params)

# data structure, units: N, mm, MPa
# Create an empty class
class varin: pass

S = varin()

S.l1 = 10. # mm
S.l2 = 50. # mm
S.lt = 2*S.l1 + S.l2
S.h1 = 10. # mm
S.h2 = 40. # mm
S.h3 = 50. # mm
S.ht = S.h1 + S.h2 + S.h3

SigC = 250 # unit: MPa

#### Propriedades de área da secção (no domínio linear elástico): centroide

Nota: a área final pode ser obtida subtraindo o retângulo maior por um retângulo menor.

In [2]:
print('big rectangular area :::::::::::::::::')
R1_yc = S.ht/2
print(f'1: yc = {R1_yc:.2f} mm')
R1_A = S.lt*S.ht
print(f'1: A = {R1_A:.2f} mm²')
R1_ycA = R1_yc*R1_A
print(f'1: (yc)A = {R1_ycA:.2f} mm³')

print('small rectangular area :::::::::::::::::')
R2_yc = S.h1 + S.h2/2
print(f'2: yc = {R2_yc:.2f} mm')
R2_A = S.l2*S.h2
print(f'2: A = {R2_A:.2f} mm²')
R2_ycA = R2_yc*R2_A
print(f'2: (yc)A = {R2_ycA:.2f} mm³')

print('\nArea of the cross-section :::::::::::::::::')
S.A = R1_A - R2_A
print(f'S: Area = {S.A:.2f} mm²')
print('\ny centroid of the cross-section :::::::::::::::::')
S.Yc = (R1_ycA - R2_ycA)/S.A
print(f'S: ycent. = {S.Yc:.2f} mm')


big rectangular area :::::::::::::::::
1: yc = 50.00 mm
1: A = 7000.00 mm²
1: (yc)A = 350000.00 mm³
small rectangular area :::::::::::::::::
2: yc = 30.00 mm
2: A = 2000.00 mm²
2: (yc)A = 60000.00 mm³

Area of the cross-section :::::::::::::::::
S: Area = 5000.00 mm²

y centroid of the cross-section :::::::::::::::::
S: ycent. = 58.00 mm


##### Propriedades de área da secção (no domínio linear elástico): momento de inércia de área

In [15]:
def irect(b,h): return b*h**3/12

print('\nTeor. Eixos Paralelos: R1 :::::::::::::::::')
d1 = S.Yc - S.ht/2
R1_I = irect(S.lt,S.ht) + R1_A*d1**2
print(f'1: d1 = {d1:.1f} mm | Iz1 = {R1_I:.2f} mm⁴')

print('\nTeor. Eixos Paralelos: R2 :::::::::::::::::')
d2 = S.Yc - (S.h1 + S.h2/2)
R2_I = irect(S.l2,S.h2) + R2_A*d2**2
print(f'2: d2 = {d2:.1f} mm | Iz2 = {R2_I:.2f} mm⁴')

print('\nMomento inercia seccao Iz ::::::::::::::::::::')
S.Iz = R1_I - R2_I
print(f'S.Iz = {S.Iz:.2f} = {S.Iz:.3e} mm⁴')

print('\nMomento inercia seccao Iy (complemento) ::::::::::::::::::::')
S.Iy = irect(S.ht,S.lt) - irect(S.h2,S.l2)
print(f'S.Iy = {S.Iy:.2f} mm⁴ = = {S.Iy*1e-12:.4e} m⁴')


Teor. Eixos Paralelos: R1 :::::::::::::::::
1: d1 = 8.0 mm | Iz1 = 6281333.33 mm⁴

Teor. Eixos Paralelos: R2 :::::::::::::::::
2: d2 = 28.0 mm | Iz2 = 1834666.67 mm⁴

Momento inercia seccao Iz ::::::::::::::::::::
S.Iz = 4446666.67 mm⁴

Momento inercia seccao Iy (complemento) ::::::::::::::::::::
S.Iy = 2441666.67 mm⁴ = = 2.4417e-06 m⁴


- Momento fletor aplicado, correspondente à tensão de cedência do matarial

Esquema da configuração e distribuição da tensão normal no instante de cedência:

<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSII/Notebooks/Au03/P1/MSII_Au03_P1a.png"
width="300">

A fórmula de Euler pode ser usada, neste cenário, para determinar o momento fletor elástico máximo, correspondente à tensão de cedência do material, 

\begin{equation*}
|\sigma_{xx}^\textrm{max}| = \Big|- \frac{M_z}{I_z} y_\mathrm{max} \Big|
\quad\therefore\quad
M_\mathrm{C} = \frac{\sigma_\mathrm{ced}I_z}{y_\mathrm{max}}  
\end{equation*}

In [4]:
ymax = np.max([S.ht - S.Yc,np.abs(-S.Yc)])
print(f'ymax = {ymax:.1f} mm')
Mc = SigC*S.Iz/ymax

print(f'Mc = {SigC:.2f}({S.Iz:.3e})/{ymax:.2f} = {Mc:.2f} N.mm = {Mc*1e-3:.2f} N.m')

ymax = 58.0 mm
Mc = 250.00(4.447e+06)/58.00 = 19166666.67 N.mm = 19166.67 N.m


## Momento Plástico

O momento plástico é considerado o memento fletor a aplicar na viga que conduz à uma plastificação completa do material ao longo da secção transversal da viga, assumindo um material elasto-plastico perfeito.

### O que acontece à posição do eixo neutro na transição entre o regime elástico e o plástico?

 Em secções que não sejam simétricas em relação ao eixo segundo o qual atua o momento fletor, **o eixo neutro passa a ser o eixo que divide a secção em duas partes com área igual**. Esta condição resulta diretamente da análise do equilíbrio estático, em que a distribuição da tensão normal deve igualar uma força axial externa nula:

\begin{equation*}
F_x = 0 \quad\therefore\quad
F_x = \int_A \sigma_{xx}\mathrm{d}A\;\Big|_{\text{ced}} = \int_{A_\mathrm{T}} \sigma_\mathrm{ced}\mathrm{d}A +  \int_{A_\mathrm{C}} (-\sigma_{\mathrm{ced}})\mathrm{d}A
= \sigma_\mathrm{ced}(A_\mathrm{T}-A_\mathrm{C}) = 0
\end{equation*}

resultando,

\begin{equation*}
A_T  = A_C \equiv A_p \qquad (~\wedge\quad\sigma_\mathrm{ced} \ne 0~)
\end{equation*}

Ou seja, as áreas da secção transversal sujeitas a uma tensão de tração e uma tensão de compressão devem ser iguais.

Por outro lado, o momento fletor plástico máximo deverá satisfazer a seguinte equação de equilíbrio:

\begin{equation*}
M_p = -\int_A y\sigma_{xx}\mathrm{d}A = -\int_{A_\mathrm{T}} y\sigma_\mathrm{ced}\mathrm{d}A -  \int_{A_\mathrm{C}} y(-\sigma_{\mathrm{ced}})\mathrm{d}A
\end{equation*}

resultando,

\begin{equation*}
M_p = \sigma_\mathrm{ced}(\overline{y}_\mathrm{C} A_\mathrm{C} - \overline{y}_\mathrm{T} A_\mathrm{T})
= \sigma_\mathrm{ced}A_p(\overline{y}_\mathrm{C} - \overline{y}_\mathrm{T})
\end{equation*}

onde $\overline{y}_\mathrm{T}$ e $\overline{y}_\mathrm{C}$ são as coordenadas dos centróides de $A_\mathrm{T}$ e $A_\mathrm{C}$ em relação ao eixo neutro. Note-se que $\overline{y}_\mathrm{T}$ é geralmente negativo, uma vez que é medido em relação à origem do sistema de coordenadas coincidente com o centroide da secção. Resulta assim:

\begin{equation*}
M_p = \left. R_1 \cdot \overline{y}_\mathrm{C} \right|_\textrm{região superior (C)} + \left. R_2 \cdot \overline{y}_\mathrm{T} \right|_\textrm{região inferior (T)}
\end{equation*}

In [5]:
print(f'S: Area = {S.A:.2f} mm²')

S: Area = 5000.00 mm²


- Divisão da área de secção em partes iguais

\begin{equation*}
A_\textrm{C} + A_\textrm{T} = A
\quad\wedge\quad
A_\textrm{C} = A_\textrm{T} \equiv A_p
\quad\wedge\quad
 A_p = \frac{A}{2}
\end{equation*}

In [6]:
P_Ap = S.A/2
print(f'S: Area/2 (plasticidade)  = {P_Ap:.2f} mm²')

S: Area/2 (plasticidade)  = 2500.00 mm²


- Cálculo da localização do Eixo Neutro em plasticidade

Considerando a distância vertical $y_p$, medida em relação ao topo da secção correspondente a uma área retangular:

\begin{equation*}
A_p = l_t \cdot y_p
\quad\therefore\quad
y_p = \frac{A_p}{l_t}
\quad\textrm{[m]}
\end{equation*}

In [7]:
P_Yp = P_Ap/S.lt
print(f'S.Yp (plasticidade) = {P_Ap}/{S.lt} = {P_Yp:.2f} mm')

S.Yp (plasticidade) = 2500.0/70.0 = 35.71 mm


O momento plástico deve ser calculado em relação a esse eixo que dista 35.71 mm em relação ao topo da secção de 100 mm de altura. A equação de base expressa o equilíbrio de forças internas resultantes:
 
\begin{equation*}
M_p = R_1 \cdot d_1 + R_2 \cdot d_2
\end{equation*}

É necessário agora calcular a resultante as forças ($R_1$, $R_2$) e o seu ponto de aplicação (centróides das sub-regiões da secção).

- Esquema da distribuição da tensão normal no instante de plastificação total de uma secção:

<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSII/Notebooks/Au03/P1/MSII_Au03_P1b.png"
width="300">

- Região superior (compressão)

A distribuição uniforme das tensões de cedência na região acima do eixo neutro tem uma força resultante dada por

\begin{equation*}
R_1 = R_2 = R = \sigma_\mathrm{ced} \frac{A}{2} = \sigma_\mathrm{ced} A_p
\end{equation*}

Esta força resultante deve ser aplicada no centróide da secção.

In [8]:
P_res = SigC*P_Ap
print(f'Resultante (tracao)  = {P_res:.2f} N')

Resultante (tracao)  = 625000.00 N


A distribuição das tensões é equivalente à aplicação da força resultante no centroide da área correspondente. Esta força exerce um momento fletor em relação ao eixo neutro (que divida a secção total em áreas iguais). É preciso ainda determinar a distância, $d_{p1}$, para o cálculo do momento flector resultante, entre o centroide da sub-área e o eixo neutro.

\begin{equation*}
d_{p1} = \frac{y_p}{2}
\end{equation*}

In [9]:
P_d1 =  P_Yp/2
print(f'd1.plastico (tracao) = {P_Yp:.2f}/2 = {P_d1:.2f} mm')

d1.plastico (tracao) = 35.71/2 = 17.86 mm


- Região inferior (tração)

O cálculo da resultante das forças e o seu ponto de aplicação, correspondente à região inferior, deve ser agora cálculado.
Para calcular $d_{p2}$ é necessário determinar a posição do centro geométrico da figura inferior (região de tração).

In [10]:
print('larger rectangular area:::::::::::::::::')
P_R1_yc = (S.ht-P_Yp)/2
P_R1_A = S.lt*(S.ht-P_Yp)
P_R1_ycA = P_R1_yc*P_R1_A
print(f'1: yc = {P_R1_yc:.2f} mm | A = {P_R1_A:.2f} mm² | (yc)A = {P_R1_ycA:.2f} mm³')

print('small rectangular area :::::::::::::::::')
P_R2_yc = S.h1 + S.h2/2
P_R2_A = S.l2*S.h2
P_R2_ycA = P_R2_yc*P_R2_A
print(f'2: yc = {P_R2_yc:.2f} mm | A = {P_R2_A:.2f} mm² | (yc)A = {P_R2_ycA:.2f} mm³')

print('\nArea of the cross-section :::::::::::::::::')
P_A = P_R1_A - P_R2_A
print(f'S: Area = {P_A:.2f} mm²')
print('\ny centroid of the cross-section :::::::::::::::::')
P_Yc2 = (P_R1_ycA - P_R2_ycA)/P_A
print(f'ycent. (região de tração) = {P_Yc2:.2f} mm')

larger rectangular area:::::::::::::::::
1: yc = 32.14 mm | A = 4500.00 mm² | (yc)A = 144642.86 mm³
small rectangular area :::::::::::::::::
2: yc = 30.00 mm | A = 2000.00 mm² | (yc)A = 60000.00 mm³

Area of the cross-section :::::::::::::::::
S: Area = 2500.00 mm²

y centroid of the cross-section :::::::::::::::::
ycent. (região de tração) = 33.86 mm


In [11]:
print('\nd2 distance :::::::::::::::::')
P_d2 = S.ht - P_Yp - P_Yc2
print(f'd2 (tração) = {S.ht:.1f} - {P_Yp:.2f} - {P_Yc2:.2f} = {P_d2:.2f} mm')


d2 distance :::::::::::::::::
d2 (tração) = 100.0 - 35.71 - 33.86 = 30.43 mm


Finalmente, o momento fletor aplicado que conduz à plastificação da secção deve ser equilibrado internamente pelo momento gerado pelo par de forças resultantes da distribuição das tensões na secção superior e inferior ao eixo neutro,

\begin{equation*}
M_p = R_1\cdot d_1 + R_2\cdot d_2
= \sigma_\mathrm{c}\frac{A}{2}(d_1+d_2)
\end{equation*}

In [12]:
P_Mom = P_res*(P_d1 + P_d2)
print(f' Momento plãsticom, Mp = {P_res:.1f}({P_d1:.2f} + {P_d2:.2f}) = {P_res:.1f}({P_d1+P_d2:.2f})')
print(f'                    Mp = {P_Mom:.2f} N.mm = {P_Mom*1e-3:.2f} N.m')

 Momento plãsticom, Mp = 625000.0(17.86 + 30.43) = 625000.0(48.29)
                    Mp = 30178571.43 N.mm = 30178.57 N.m


No contexto de flexão de vigas, o fator de forma $𝐾$ é definido como a razão entre o momento plástico ,

\begin{equation*}
K = \frac{M_p}{M_c}
\end{equation*}

In [13]:
K = Mc/P_Mom
print(f' K = {P_Mom/Mc:.2f}')

 K = 1.57


Este fator indica o quanto de momento adicional uma viga pode suportar após atingir o limite elástico, entrando na região plástica.

Por exemplo, para a secção dada o valor de $𝐾 = 1,57$ significa que o momento plástico é aproximadamente 57% maior do que o momento elástico.

O fator de forma é útil na análise plástica e em projeto, pois ajuda a perceber a capacidade estrutural adicional de uma viga após o início da cedência.

---

Copyright (c) DEMI - FCT NOVA

Interactive computing by <a href="https://jupyter.org/" target="_blank"> <span
style="color:#333399"> Jupyter Notebook </span> </a> &nbsp;|&nbsp;Coded by <a href = "mailto: jmc.xavier@fct.unl.pt">José Xavier</a>

Licensed under  <a href="http://creativecommons.org/licenses/by-sa/4.0/"
target="_blank"> <span style="color:#333399;font-size: 20px"> CC BY-SA 4.0  </span></a>